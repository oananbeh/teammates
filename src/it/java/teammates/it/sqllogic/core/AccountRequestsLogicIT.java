package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.AccountRequestStatus;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountRequestsLogic;
import teammates.storage.sqlapi.AccountRequestsDb;
import teammates.storage.sqlentity.AccountRequest;

/**
 * SUT: {@link AccountRequestsLogic}.
 */
public class AccountRequestsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountRequestsLogic accountRequestsLogic = AccountRequestsLogic.inst();

    @Test
    public void testGetAccountRequest_nonExistentAccountRequest_returnsNull() {
        UUID id = UUID.randomUUID();
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertNull(actualAccountRequest);
    }

    @Test
    public void testGetAccountRequest_existingAccountRequest_getsSuccessfully() throws InvalidParametersException {
        AccountRequest expectedAccountRequest =
                new AccountRequest("test@gmail.com", "name", "institute", AccountRequestStatus.PENDING, "comments");
        UUID id = expectedAccountRequest.getId();
        accountRequestsLogic.createAccountRequest(expectedAccountRequest);
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertEquals(expectedAccountRequest, actualAccountRequest);
    }

    @Test
    public void testResetAccountRequest()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {

        ______TS("success: create account request and update registeredAt field");

        String name = "name lee";
        String email = "email@gmail.com";
        String institute = "institute";
        AccountRequestStatus status = AccountRequestStatus.PENDING;
        String comments = "comments";

        AccountRequest toReset = accountRequestsLogic.createAccountRequest(name, email, institute, status, comments);
        AccountRequestsDb accountRequestsDb = AccountRequestsDb.inst();

        toReset.setRegisteredAt(Instant.now());
        UUID id = toReset.getId();
        toReset = accountRequestsDb.getAccountRequest(id);

        assertNotNull(toReset);
        assertNotNull(toReset.getRegisteredAt());

        ______TS("success: reset account request that already exists");

        AccountRequest resetted = accountRequestsLogic.resetAccountRequest(id);

        assertNull(resetted.getRegisteredAt());

        ______TS("success: test delete account request");

        accountRequestsLogic.deleteAccountRequest(toReset.getId());

        assertNull(accountRequestsLogic.getAccountRequest(toReset.getId()));

        ______TS("failure: reset account request that does not exist");

        assertThrows(EntityDoesNotExistException.class,
                () -> accountRequestsLogic.resetAccountRequest(id));
    }
}
