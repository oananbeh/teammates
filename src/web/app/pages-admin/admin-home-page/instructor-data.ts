/**
 * Represents the data of a new instructor.
 */
export interface InstructorData {
  name: string;
  email: string;
  institution: string;
  status: string;
  isCurrentlyBeingEdited: boolean;
  statusCode?: number;
  joinLink?: string;
  message?: string;
}
