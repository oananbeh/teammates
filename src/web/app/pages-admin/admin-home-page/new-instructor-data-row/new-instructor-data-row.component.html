<td>
  <span *ngIf="!isBeingEdited" id="instructor-{{ index }}-name">{{ instructor.name }}</span>
  <input *ngIf="isBeingEdited" class="form-control" type="text" id="instructor-{{ index }}-name-input" [(ngModel)]="editedInstructorName">
</td>
<td>
  <span *ngIf="!isBeingEdited" id="instructor-{{ index }}-email">{{ instructor.email }}</span>
  <input *ngIf="isBeingEdited" class="form-control" type="text" id="instructor-{{ index }}-email-input" [(ngModel)]="editedInstructorEmail">
</td>
<td>
  <span *ngIf="!isBeingEdited" id="instructor-{{ index }}-institution">{{ instructor.institution }}</span>
  <input *ngIf="isBeingEdited" class="form-control" type="text" id="instructor-{{ index }}-institution-input" [(ngModel)]="editedInstructorInstitution">
</td>
<td style="width: 17%;">
  <div class="button-group" *ngIf="isBeingEdited && (instructor.status === 'PENDING' || instructor.status === 'FAIL')">
    <button class="action-btn btn btn-sm btn-primary text-nowrap" id="confirm-edit-instructor-{{ index }}" style="margin-right: 10px;" (click)="confirmEditInstructor()">Confirm</button>
    <button class="action-btn btn btn-sm btn-danger text-nowrap" id="cancel-edit-instructor-{{ index }}" style="margin-right: 10px;" (click)="cancelEditInstructor()">Cancel</button>
  </div>
  <div class="button-group" *ngIf="!isBeingEdited && (instructor.status === 'PENDING' || instructor.status === 'FAIL')">
    <button class="action-btn btn btn-sm btn-primary text-nowrap" id="add-instructor-{{ index }}" style="margin-right: 10px;" (click)="addInstructor()" [disabled]="isAddDisabled">Add</button>
    <button class="action-btn btn btn-sm btn-info text-nowrap" id="edit-instructor-{{ index }}" style="margin-right: 10px;" (click)="editInstructor()">Edit</button>
    <button class="action-btn btn btn-sm btn-danger" id="remove-instructor-{{ index }}" (click)="removeInstructor()">Remove</button>
  </div>
</td>
<td style="width: 8%;">{{ instructor.status }}</td>
<td style="width: 40%;">
  <span *ngIf="instructor.status === 'SUCCESS'" id="message-instructor-{{ index }}">
    Instructor "{{ instructor.name }}" has been successfully created. [<a href="{{ instructor.joinLink }}">join link</a>]
  </span>
  <span class="message" *ngIf="instructor.status === 'FAIL'" id="message-instructor-{{ index }}">{{ instructor.message }}</span>
</td>
