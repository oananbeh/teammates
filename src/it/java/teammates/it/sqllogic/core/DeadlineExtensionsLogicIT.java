package teammates.it.sqllogic.core;

import java.time.Instant;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import teammates.common.datatransfer.SqlDataBundle;
import teammates.common.util.HibernateUtil;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.DeadlineExtensionsLogic;
import teammates.storage.sqlentity.FeedbackSession;
import teammates.storage.sqlentity.Student;

/**
 * SUT: {@link DeadlineExtensionsLogic}.
 */
public class DeadlineExtensionsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private DeadlineExtensionsLogic deadlineExtensionsLogic = DeadlineExtensionsLogic.inst();
    private SqlDataBundle typicalDataBundle;

    @Override
    @BeforeClass
    public void setupClass() {
        super.setupClass();
        typicalDataBundle = getTypicalSqlDataBundle();
    }

    @Override
    @BeforeMethod
    protected void setUp() throws Exception {
        super.setUp();
        persistDataBundle(typicalDataBundle);
        HibernateUtil.flushSession();
        HibernateUtil.clearSession();
    }

    @Test
    public void testGetExtendedDeadline_extensionExists_success() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        assert student != null;
        Instant extendedDeadlineForStudent = deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student);

        assertNotNull(extendedDeadlineForStudent);
        assertEquals(Instant.parse("2027-04-30T23:00:00Z"), extendedDeadlineForStudent);
    }

    @Test
    public void testGetExtendedDeadline_extensionDoesNotExist_null() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student2InCourse1");
        Instant extendedDeadlineForStudent = deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student);

        assertNull(extendedDeadlineForStudent);
    }
}
