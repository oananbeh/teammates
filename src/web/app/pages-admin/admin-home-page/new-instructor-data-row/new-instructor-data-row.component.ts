import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { InstructorData } from '../instructor-data';

/**
 * A single row of data of a new instructor.
 */
@Component({
  // The following selector code style violation of https://angular.io/guide/styleguide#style-05-02 and
  // https://angular.io/guide/styleguide#style-05-03 seems necessary according to
  // https://stackoverflow.com/questions/55446740/how-to-add-row-component-in-table-in-angular-7
  // eslint-disable-next-line @angular-eslint/component-selector
  selector: 'tr[tm-new-instructor-data-row]',
  templateUrl: './new-instructor-data-row.component.html',
  styleUrls: ['./new-instructor-data-row.component.scss'],
})
export class NewInstructorDataRowComponent implements OnInit {
  @Input() instructor!: InstructorData;
  @Input() index!: number;
  @Input() isAddDisabled!: boolean;
  @Output() addInstructorEvent: EventEmitter<void> = new EventEmitter();
  @Output() removeInstructorEvent: EventEmitter<void> = new EventEmitter();
  @Output() toggleEditModeEvent: EventEmitter<boolean> = new EventEmitter();

  isBeingEdited: boolean = false;
  editedInstructorName!: string;
  editedInstructorEmail!: string;
  editedInstructorInstitution!: string;

  ngOnInit(): void {
    this.resetEditedInstructorDetails();
  }

  addInstructor(): void {
    this.addInstructorEvent.emit();
  }

  removeInstructor(): void {
    this.removeInstructorEvent.emit();
  }

  /**
   * Starts editing the instructor.
   */
  editInstructor(): void {
    this.setEditModeAndAlertParent(true);
  }

  /**
   * Confirms the edit of the instructor's details.
   */
  confirmEditInstructor(): void {
    this.instructor.name = this.editedInstructorName;
    this.instructor.email = this.editedInstructorEmail;
    this.instructor.institution = this.editedInstructorInstitution;
    this.setEditModeAndAlertParent(false);
  }

  /**
   * Cancels the edit of the instructor's details.
   */
  cancelEditInstructor(): void {
    this.setEditModeAndAlertParent(false);
    this.resetEditedInstructorDetails();
  }

  /**
   * Resets the edited instructor details to the original details.
   */
  private resetEditedInstructorDetails(): void {
    this.editedInstructorName = this.instructor.name;
    this.editedInstructorEmail = this.instructor.email;
    this.editedInstructorInstitution = this.instructor.institution;
  }

  /**
   * Sets whether edit mode is enabled and then alerts the parent of the new status.
   *
   * @param isEnabled Whether edit mode is enabled.
   */
  private setEditModeAndAlertParent(isEnabled: boolean): void {
    this.isBeingEdited = isEnabled;
    this.alertParentEditModeToggled();
  }

  /**
   * Alerts the parent that the edit mode was toggled and passes whether this is in edit mode or not.
   */
  private alertParentEditModeToggled(): void {
    this.toggleEditModeEvent.emit(this.isBeingEdited);
  }

}
