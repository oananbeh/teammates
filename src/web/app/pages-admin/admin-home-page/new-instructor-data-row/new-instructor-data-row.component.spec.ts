import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { NewInstructorDataRowComponent } from './new-instructor-data-row.component';
import testEventEmission from '../../../../test-helpers/test-event-emitter';
import { InstructorData } from '../instructor-data';

describe('NewInstructorDataRowComponent', () => {
  let component: NewInstructorDataRowComponent;
  let fixture: ComponentFixture<NewInstructorDataRowComponent>;
  let expectedInstructorData: InstructorData;
  let expectedIndex: number;
  let expectedIsAddDisabled: boolean;

  let addButtonDe: any;
  let addButtonEl: any;
  let editButtonDe: any;
  let editButtonEl: any;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule],
      declarations: [NewInstructorDataRowComponent],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(NewInstructorDataRowComponent);
    component = fixture.componentInstance;
    expectedInstructorData = {
      name: 'Instructor',
      email: 'instructor@instruct.or',
      institution: 'Institutional Institution of Institute',
      status: 'PENDING',
      isCurrentlyBeingEdited: false,
    };
    expectedIndex = 0;
    expectedIsAddDisabled = false;
    component.instructor = expectedInstructorData;
    component.index = expectedIndex;
    component.isAddDisabled = expectedIsAddDisabled;
    fixture.detectChanges();
    addButtonDe = fixture.debugElement
      .query(By.css(`#add-instructor-${expectedIndex}`));
    addButtonEl = addButtonDe.nativeElement;
    editButtonDe = fixture.debugElement
      .query(By.css(`#edit-instructor-${expectedIndex}`));
    editButtonEl = editButtonDe.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();

    expect(component.isBeingEdited).toBeFalsy();

    expect(component.editedInstructorName).toEqual(expectedInstructorData.name);
    expect(component.editedInstructorEmail).toEqual(expectedInstructorData.email);
    expect(component.editedInstructorInstitution).toEqual(expectedInstructorData.institution);
  });

  it('should display the instructor name received as input', () => {
    const displayedInstructorName: string = fixture.debugElement
      .query(By.css(`#instructor-${expectedIndex}-name`))
      .nativeElement.textContent;
    expect(displayedInstructorName).toEqual(expectedInstructorData.name);
  });

  it('should display the instructor email received as input', () => {
    const displayedInstructorEmail: string = fixture.debugElement
      .query(By.css(`#instructor-${expectedIndex}-email`))
      .nativeElement.textContent;
    expect(displayedInstructorEmail).toEqual(expectedInstructorData.email);
  });

  it('should display the instructor institution received as input', () => {
    const displayedInstructorInstitution: string = fixture.debugElement
      .query(By.css(`#instructor-${expectedIndex}-institution`))
      .nativeElement.textContent;
    expect(displayedInstructorInstitution).toEqual(expectedInstructorData.institution);
  });

  it('should have an add button that is not disabled when isAddDisabled is false', () => {
    expect(addButtonEl.disabled).toBeFalsy();
  });

  it('should have an add button that is disabled when isAddDisabled is true', () => {
    component.isAddDisabled = true;
    fixture.detectChanges();
    expect(addButtonEl.disabled).toBeTruthy();
  });

  it('should emit addInstructorEvent when adding', () => {
    let hasEmitted: boolean = false;
    testEventEmission(component.addInstructorEvent, () => { hasEmitted = true; });

    addButtonDe.triggerEventHandler('click', null);
    expect(hasEmitted).toBeTruthy();
  });

  it('should emit removeInstructorEvent when removing', () => {
    let hasEmitted = false;
    testEventEmission(component.removeInstructorEvent, () => { hasEmitted = true; });

    fixture.debugElement
      .query(By.css(`#remove-instructor-${expectedIndex}`))
      .triggerEventHandler('click', null);
    expect(hasEmitted).toBeTruthy();
  });

  it('should emit true via toggleEditModeEvent when entering edit mode', () => {
    let isInEditMode: boolean | undefined;
    testEventEmission(component.toggleEditModeEvent, (emittedValue) => { isInEditMode = emittedValue; });

    editButtonDe.triggerEventHandler('click', null);
    expect(isInEditMode).toBeTruthy();
  });

  it('should emit false via toggleEditModeEvent when confirming the edit', () => {
    let isInEditMode: boolean | undefined;
    testEventEmission(component.toggleEditModeEvent, (emittedValue) => { isInEditMode = emittedValue; }, false);

    editButtonDe.triggerEventHandler('click', null);
    fixture.detectChanges();

    fixture.debugElement
      .query(By.css(`#confirm-edit-instructor-${expectedIndex}`))
      .triggerEventHandler('click', null);
    expect(isInEditMode).toBeFalsy();
  });

  it('should emit false via toggleEditModeEvent when cancelling the edit', () => {
    let isInEditMode: boolean | undefined;
    testEventEmission(component.toggleEditModeEvent, (emittedValue) => { isInEditMode = emittedValue; }, false);

    editButtonDe.triggerEventHandler('click', null);
    fixture.detectChanges();

    fixture.debugElement
      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
      .triggerEventHandler('click', null);
    expect(isInEditMode).toBeFalsy();
  });

  it('should set isBeingEdited to true when editing starts', () => {
    editButtonEl.click();

    expect(component.isBeingEdited).toBeTruthy();
  });

  it('should set isBeingEdited to false when the edit is confirmed', () => {
    editButtonEl.click();
    fixture.detectChanges();

    const confirmButtonEl: any = fixture.debugElement
      .query(By.css(`#confirm-edit-instructor-${expectedIndex}`))
      .nativeElement;
    confirmButtonEl.click();

    expect(component.isBeingEdited).toBeFalsy();
  });

  it('should set isBeingEdited to false when the edit is cancelled', () => {
    editButtonEl.click();
    fixture.detectChanges();

    const cancelButtonEl: any = fixture.debugElement
      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
      .nativeElement;
    cancelButtonEl.click();

    expect(component.isBeingEdited).toBeFalsy();
  });

  it('should initially have the unedited instructor details when editing starts', () => {
    editButtonEl.click();

    expect(component.editedInstructorName).toEqual(expectedInstructorData.name);
    expect(component.editedInstructorEmail).toEqual(expectedInstructorData.email);
    expect(component.editedInstructorInstitution).toEqual(expectedInstructorData.institution);
  });

  it('should update the instructor details when the edit is confirmed', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    const confirmButtonEl: any = fixture.debugElement
      .query(By.css(`#confirm-edit-instructor-${expectedIndex}`))
      .nativeElement;
    confirmButtonEl.click();

    expect(component.instructor.name).toEqual('Edited Name');
    expect(component.instructor.email).toEqual('Edited@ema.il');
    expect(component.instructor.institution).toEqual('Edited Institution');
  });

  it('should not update the instructor details when the edit is cancelled', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    const cancelButtonEl: any = fixture.debugElement
      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
      .nativeElement;
    cancelButtonEl.click();

    expect(component.instructor.name).toEqual(expectedInstructorData.name);
    expect(component.instructor.email).toEqual(expectedInstructorData.email);
    expect(component.instructor.institution).toEqual(expectedInstructorData.institution);
  });

  it('should reset the edited instructor details when the edit is cancelled', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    const cancelButtonEl: any = fixture.debugElement
      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
      .nativeElement;
    cancelButtonEl.click();

    expect(component.editedInstructorName).toEqual(expectedInstructorData.name);
    expect(component.editedInstructorEmail).toEqual(expectedInstructorData.email);
    expect(component.editedInstructorInstitution).toEqual(expectedInstructorData.institution);
  });

  it('should initially have the original instructor details when editing starts after a cancellation', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    const cancelButtonEl: any = fixture.debugElement
      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
      .nativeElement;
    cancelButtonEl.click();
    fixture.detectChanges();

    editButtonEl.click();

    expect(component.editedInstructorName).toEqual(expectedInstructorData.name);
    expect(component.editedInstructorEmail).toEqual(expectedInstructorData.email);
    expect(component.editedInstructorInstitution).toEqual(expectedInstructorData.institution);
  });

  it('should snap with default view', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with start of edit', () => {
    editButtonEl.click();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should snap with instructor details edited', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should snap with edit confirmed', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    const confirmButtonEl: any = fixture.debugElement
      .query(By.css(`#confirm-edit-instructor-${expectedIndex}`))
      .nativeElement;
    confirmButtonEl.click();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should snap with edit cancelled', () => {
    editButtonEl.click();
    fixture.detectChanges();

    component.editedInstructorName = 'Edited Name';
    component.editedInstructorEmail = 'Edited@ema.il';
    component.editedInstructorInstitution = 'Edited Institution';
    const cancelButtonEl: any = fixture.debugElement
      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
      .nativeElement;
    cancelButtonEl.click();
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
