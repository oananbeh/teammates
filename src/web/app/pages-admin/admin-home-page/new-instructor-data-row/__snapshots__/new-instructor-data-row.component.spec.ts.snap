// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewInstructorDataRowComponent should snap with default view 1`] = `
<tr
  addInstructorEvent={[Function EventEmitter_]}
  editedInstructorEmail={[Function String]}
  editedInstructorInstitution={[Function String]}
  editedInstructorName={[Function String]}
  index="0"
  instructor={[Function Object]}
  isAddDisabled="false"
  isBeingEdited="false"
  removeInstructorEvent={[Function EventEmitter_]}
  toggleEditModeEvent={[Function EventEmitter_]}
>
  <td>
    <span
      id="instructor-0-name"
    >
      Instructor
    </span>
  </td><td>
    <span
      id="instructor-0-email"
    >
      instructor@instruct.or
    </span>
  </td><td>
    <span
      id="instructor-0-institution"
    >
      Institutional Institution of Institute
    </span>
  </td><td
    style="width: 17%;"
  >
    <div
      class="button-group"
    >
      <button
        class="action-btn btn btn-sm btn-primary text-nowrap"
        id="add-instructor-0"
        style="margin-right: 10px;"
      >
        Add
      </button>
      <button
        class="action-btn btn btn-sm btn-info text-nowrap"
        id="edit-instructor-0"
        style="margin-right: 10px;"
      >
        Edit
      </button>
      <button
        class="action-btn btn btn-sm btn-danger"
        id="remove-instructor-0"
      >
        Remove
      </button>
    </div>
  </td><td
    style="width: 8%;"
  >
    PENDING
  </td><td
    style="width: 40%;"
  >
  </td>
</tr>
`;

exports[`NewInstructorDataRowComponent should snap with edit cancelled 1`] = `
<tr
  addInstructorEvent={[Function EventEmitter_]}
  editedInstructorEmail={[Function String]}
  editedInstructorInstitution={[Function String]}
  editedInstructorName={[Function String]}
  index="0"
  instructor={[Function Object]}
  isAddDisabled="false"
  isBeingEdited="false"
  removeInstructorEvent={[Function EventEmitter_]}
  toggleEditModeEvent={[Function EventEmitter_]}
>
  <td>
    <span
      id="instructor-0-name"
    >
      Instructor
    </span>
  </td><td>
    <span
      id="instructor-0-email"
    >
      instructor@instruct.or
    </span>
  </td><td>
    <span
      id="instructor-0-institution"
    >
      Institutional Institution of Institute
    </span>
  </td><td
    style="width: 17%;"
  >
    <div
      class="button-group"
    >
      <button
        class="action-btn btn btn-sm btn-primary text-nowrap"
        id="add-instructor-0"
        style="margin-right: 10px;"
      >
        Add
      </button>
      <button
        class="action-btn btn btn-sm btn-info text-nowrap"
        id="edit-instructor-0"
        style="margin-right: 10px;"
      >
        Edit
      </button>
      <button
        class="action-btn btn btn-sm btn-danger"
        id="remove-instructor-0"
      >
        Remove
      </button>
    </div>
  </td><td
    style="width: 8%;"
  >
    PENDING
  </td><td
    style="width: 40%;"
  >
  </td>
</tr>
`;

exports[`NewInstructorDataRowComponent should snap with edit confirmed 1`] = `
<tr
  addInstructorEvent={[Function EventEmitter_]}
  editedInstructorEmail={[Function String]}
  editedInstructorInstitution={[Function String]}
  editedInstructorName={[Function String]}
  index="0"
  instructor={[Function Object]}
  isAddDisabled="false"
  isBeingEdited="false"
  removeInstructorEvent={[Function EventEmitter_]}
  toggleEditModeEvent={[Function EventEmitter_]}
>
  <td>
    <span
      id="instructor-0-name"
    >
      Edited Name
    </span>
  </td><td>
    <span
      id="instructor-0-email"
    >
      Edited@ema.il
    </span>
  </td><td>
    <span
      id="instructor-0-institution"
    >
      Edited Institution
    </span>
  </td><td
    style="width: 17%;"
  >
    <div
      class="button-group"
    >
      <button
        class="action-btn btn btn-sm btn-primary text-nowrap"
        id="add-instructor-0"
        style="margin-right: 10px;"
      >
        Add
      </button>
      <button
        class="action-btn btn btn-sm btn-info text-nowrap"
        id="edit-instructor-0"
        style="margin-right: 10px;"
      >
        Edit
      </button>
      <button
        class="action-btn btn btn-sm btn-danger"
        id="remove-instructor-0"
      >
        Remove
      </button>
    </div>
  </td><td
    style="width: 8%;"
  >
    PENDING
  </td><td
    style="width: 40%;"
  >
  </td>
</tr>
`;

exports[`NewInstructorDataRowComponent should snap with instructor details edited 1`] = `
<tr
  addInstructorEvent={[Function EventEmitter_]}
  editedInstructorEmail={[Function String]}
  editedInstructorInstitution={[Function String]}
  editedInstructorName={[Function String]}
  index="0"
  instructor={[Function Object]}
  isAddDisabled="false"
  isBeingEdited={[Function Boolean]}
  removeInstructorEvent={[Function EventEmitter_]}
  toggleEditModeEvent={[Function EventEmitter_]}
>
  <td>
    <input
      class="form-control ng-untouched ng-pristine ng-valid"
      id="instructor-0-name-input"
      type="text"
    />
  </td><td>
    <input
      class="form-control ng-untouched ng-pristine ng-valid"
      id="instructor-0-email-input"
      type="text"
    />
  </td><td>
    <input
      class="form-control ng-untouched ng-pristine ng-valid"
      id="instructor-0-institution-input"
      type="text"
    />
  </td><td
    style="width: 17%;"
  >
    <div
      class="button-group"
    >
      <button
        class="action-btn btn btn-sm btn-primary text-nowrap"
        id="confirm-edit-instructor-0"
        style="margin-right: 10px;"
      >
        Confirm
      </button>
      <button
        class="action-btn btn btn-sm btn-danger text-nowrap"
        id="cancel-edit-instructor-0"
        style="margin-right: 10px;"
      >
        Cancel
      </button>
    </div>
  </td><td
    style="width: 8%;"
  >
    PENDING
  </td><td
    style="width: 40%;"
  >
  </td>
</tr>
`;

exports[`NewInstructorDataRowComponent should snap with start of edit 1`] = `
<tr
  addInstructorEvent={[Function EventEmitter_]}
  editedInstructorEmail={[Function String]}
  editedInstructorInstitution={[Function String]}
  editedInstructorName={[Function String]}
  index="0"
  instructor={[Function Object]}
  isAddDisabled="false"
  isBeingEdited={[Function Boolean]}
  removeInstructorEvent={[Function EventEmitter_]}
  toggleEditModeEvent={[Function EventEmitter_]}
>
  <td>
    <input
      class="form-control ng-untouched ng-pristine ng-valid"
      id="instructor-0-name-input"
      type="text"
    />
  </td><td>
    <input
      class="form-control ng-untouched ng-pristine ng-valid"
      id="instructor-0-email-input"
      type="text"
    />
  </td><td>
    <input
      class="form-control ng-untouched ng-pristine ng-valid"
      id="instructor-0-institution-input"
      type="text"
    />
  </td><td
    style="width: 17%;"
  >
    <div
      class="button-group"
    >
      <button
        class="action-btn btn btn-sm btn-primary text-nowrap"
        id="confirm-edit-instructor-0"
        style="margin-right: 10px;"
      >
        Confirm
      </button>
      <button
        class="action-btn btn btn-sm btn-danger text-nowrap"
        id="cancel-edit-instructor-0"
        style="margin-right: 10px;"
      >
        Cancel
      </button>
    </div>
  </td><td
    style="width: 8%;"
  >
    PENDING
  </td><td
    style="width: 40%;"
  >
  </td>
</tr>
`;
